docs/RegistryCoordinator.md

```diff
@@ -0,0 +1,325 @@
+
+The `RegistryCoordinator` has three primary functions:
+1. It is the primary entry and exit point for operators as they register for and deregister from quorums, and manages registration and deregistration in the `BLSApkRegistry`, `StakeRegistry`, and `IndexRegistry`. It also hooks into the EigenLayer core contracts, updating the core `DelegationManager` when an Operator registers/deregisters.
+2. It allows anyone to update the current stake of any registered operator
+3. It allows the Owner to initialize and configure new quorums
+
+#### High-level Concepts
+
+This document organizes methods according to the following themes (click each to be taken to the relevant section):
+* [Registering and Deregistering](#registering-and-deregistering)
+* [Updating Registered Operators](#updating-registered-operators)
+* [System Configuration](#system-configuration)
+
+#### Roles
+
+* Owner: a permissioned role that can create and configure quorums as well as manage other roles
+* Ejector: a permissioned role that can forcibly eject an operator from a quorum via `RegistryCoordinator.ejectOperator`
+* Churn Approver: a permissioned role that signs off on operation churn in `RegistryCoordinator.registerOperatorWithChurn`
+
+---    
+
+### Registering and Deregistering
+
+These methods allow operators to register for/deregister from one or more quorums, and are the primary entry points to the middleware contracts as a whole:
+* [`registerOperator`](#registeroperator)
+* [`registerOperatorWithChurn`](#registeroperatorwithchurn)
+* [`deregisterOperator`](#deregisteroperator)
+* [`ejectOperator`](#ejectoperator)
+
+#### `registerOperator`
+
+```solidity
+function registerOperator(
+    bytes calldata quorumNumbers,
+    string calldata socket,
+    SignatureWithSaltAndExpiry memory operatorSignature
+) 
+    external 
+    onlyWhenNotPaused(PAUSED_REGISTER_OPERATOR)
+```
+
+Registers the caller as an Operator for one or more quorums, as long as registration doesn't place any quorum's operator count above the configured cap. This method updates the Operator's current and historical bitmap of registered quorums, and forwards a call to each of the registry contracts:
+* `BLSApkRegistry.registerOperator`
+* `StakeRegistry.registerOperator`
+* `IndexRegistry.registerOperator`
+
+If the Operator was not currently registered for any quorums, this method will register the Operator to the AVS in the EigenLayer core contracts (`DelegationManager.registerOperatorToAVS`), passing in the provided `operatorSignature`. See the [`DelegationManager` docs][core-dmgr-docs] for more details.
+
+*Effects*:
+* If the Operator was not currently registered for any quorums: 
+    * Updates their status to `REGISTERED`
+    * Registers them in the core contracts (see [`DelegationManager.registerOperatorToAVS`][core-dmgr-register])
+* Adds the new quorums to the Operator's current registered quorums, and updates the Operator's bitmap history
```

Comment:
perhaps the first mention of bitmaps -- we might want to address these as a concept earlier on / explain the general pattern? I think I advised G to do something similar before with the way our historical storage works

Reply:
oh good call. i'll add that to the "concepts" section in the docs README

Reply:
[ac38452](https://github.com/Layr-Labs/eigenlayer-middleware/pull/96/commits/ac3845212bca0232f3db26882c12568b4a6e3660)